pkg/operator/ceph/pool/controller.go

```diff
@@ -419,64 +420,91 @@ func deletePool(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo,
 	return nil
 }
 
-// remove removes any element from a list
-func remove(list []string, s string) []string {
-	for i, v := range list {
-		if v == s {
-			list = append(list[:i], list[i+1:]...)
+// generateStatsPoolList combines existingStatsPools and rookStatsPools, removes items in removePools,
+// removes duplicates, ensures no empty strings, and returns a comma-separated string in a deterministic order.
+func generateStatsPoolList(existingStatsPools []string, rookStatsPools []string, removePools []string) string {
+	// Create a map to keep track of pools that should be removed
+	removeMap := make(map[string]struct{})
+	for _, pool := range removePools {
+		if pool != "" {
+			removeMap[pool] = struct{}{}
 		}
 	}
 
-	return list
+	// Create a list to store unique pools
+	poolList := []string{}
+
+	// Helper function to add a pool if it's not in removeMap and not already in the list
+	addUniquePool := func(pool string) {
+		if pool != "" {
+			// Check if the pool should be removed or already exists in poolList
+			if _, exists := removeMap[pool]; !exists && !contains(poolList, pool) {
+				poolList = append(poolList, pool)
+			}
+		}
```

Comment:
Go best practices are to use "happy path on the left" methods. This can be achieved by rewriting:

```suggestion
		if pool == "" {
			return
		}
		// Check if the pool should be removed or already exists in poolList
		if _, exists := removeMap[pool]; !exists && !contains(poolList, pool) {
			poolList = append(poolList, pool)
		}
```

Reply:
done